docs/content/guides/dagster/re-execution.mdx

```diff
@@ -0,0 +1,159 @@
+
+## Motivation
+
+If solids fail or upstream data has changed within a pipeline execution, the pipeline may need to be re-run starting from a particular point. Dagster calls this process `re-execution`.
+
+Imagine a machine learning pipeline with three solids, the first being the most time and resource intensive: training the model, testing the model, and building analyses on the results. If the pipeline fails with the solid that is testing the model, after fixing the root cause it would take much more time to create a new run of the pipeline than to start again from the second step.
+
+With Dagster, the re-execution of parts of the pipeline is grouped with the original run to make it easy to trace. The original pipeline execution metadata is not overwritten, making re-execution a non-destructive operation.
+
+## Example Walk-through
+
+Consider the following pipeline which has three solids, one of which fails half of the time.
+
+```python file=../../reexecution/pipeline/unreliable_pipeline.py
+import os
+
+from random import random
+from dagster import Array, Field, ModeDefinition, OutputDefinition, pipeline, solid
+from dagster.core.storage.fs_io_manager import PickledObjectFilesystemIOManager
+from dagster.core.storage.io_manager import io_manager
+
+
+class MyDatabaseIOManager(PickledObjectFilesystemIOManager):
+    def _get_path(self, context):
+        keys = context.get_run_scoped_output_identifier()
+
+        return os.path.join("/tmp", *keys)
+
+
+@io_manager(output_config_schema={"partitions": Field(Array(str), is_required=False)})
+def my_db_io_manager(_):
+    return MyDatabaseIOManager()
+
+
+@solid(
+    output_defs=[
+        OutputDefinition(io_manager_key="my_db_io_manager"),
+    ],
+)
+def unreliable_start(_):
+    return 1
+
+
+@solid(
+    output_defs=[
+        OutputDefinition(io_manager_key="my_db_io_manager"),
+    ],
+)
+def unreliable(_, num):
+    failure_rate = 0.5
+    if random() < failure_rate:
+        raise Exception("blah")
+
+
+@solid(
+    output_defs=[
+        OutputDefinition(io_manager_key="my_db_io_manager"),
+    ],
+)
+def unreliable_end(_, num):
+    return
+
+
+@pipeline(
+    mode_defs=[ModeDefinition(resource_defs={"my_db_io_manager": my_db_io_manager})],
+    description="Demo pipeline of chained solids that fail with a configurable probability."
+)
+def unreliable_pipeline():
+    unreliable_end(unreliable(unreliable_start()))
+```
+
+Although very simple, there are inputs and outputs passed between tasks. With an IO manager, re-execution is able to handle inputs and outputs stored from the initial run.
+
+Upon launching the execution in Dagit, you can find the re-execution option on the top right. You will see multiple options for different types of re-execution.
+
+\*\*\*\* INSERT NEW SCREENSHOT OF UNREABLE PIPELINE \*\*\*\*
+
+<Image
+alt="Dagit Re-execution"
+src="/images/concepts/dagit/re-execution.png"
+width={3808}
+height={2414}
+/>
+
+### Re-execution Options
```

Comment:
Maybe rename it to `Re-execution in Dagit`?

Reply:
Makes sense, updating